### DISCUSSION TITLE
How to use actionRules?

### DISCUSSION THREAD

**Discussion by JF-626164** [Power-Components/livewire-powergrid#254] [Q&A] [answered]
I read the documentation about the actionRules, I like the idea behind them.  I copied the code in the Usage section and I'm getting an "Undefined type: App\Http\Livewire\Rule" on the Rule::rows().  What class do I need to import to use this?

_Edit:_ I did some digging in the Powergrid Demo and found that the class to use is 
use PowerComponents\LivewirePowerGrid\Rules\Rule;

However, now that I'm calling that, it says "Undefined type: PowerComponents\LivewirePowerGrid\Rules\Rule"

**Comment by dansysanalyst** [comment, marked as answer]
Hi @JF-626164,

What is your PowerGrid version?

Action Rules were introduced in v2.2.0.

You can also post your code here so we can have a look!

**Reply by JF-626164** [reply]
1.5.1, I'm waaay behind lol
Updating now, I didn't realize I was so far behind.

**Reply by JF-626164** [reply]
Updated, its working now. My apologies, I regularly run composer update so I assumed I was saying up to date.

**Comment by dansysanalyst** [comment]
Welcome to new features! 🤠

Don't forget to have a look at the Upgrade guide:

https://livewire-powergrid.com/#/get-started/upgrade-v1?id=upgrade-from-v1

**Reply by JF-626164** [reply]
Thanks!

I've updated and now the Rule works, and now the DatePicker doesn't lol
The bug hunting begins
